test/amp_tags_test.js

```diff
@@ -56,12 +68,38 @@ describe('AmpTags', function () {
     assert.equal("", privateStoryResults);
   });
 
+  it("does allows you to only ampify free visual story pages", function () {
+    const story = { slug: "section/slug", "is-amp-supported": true };
+    const publicStoryResults = getSeoMetadata(
+      { ...seoConfig, ampStoryPages: "public" },
+      config,
+      "story-page",
+      { data: { story: { ...story, "story-template": "visual-story" } } },
+      {}
+    );
+    assert.equal('<link rel="amphtml" href="/section%2Fslug"/>', publicStoryResults);
+    const privateStoryResults = getSeoMetadata(
+      { ...seoConfig, ampStoryPages: "public" },
+      config,
+      "story-page",
+      { data: { story: { ...story, access: "subscription", "story-template": "visual-story" } } },
+      {}
+    );
+    assert.equal("", privateStoryResults);
+  });
+
   it("does not append domain to amp stories if appendHostToAmpUrl not present", function () {
     const story = { slug: "section/slug", "is-amp-supported": true };
     const string = getSeoMetadata(seoConfig, config, "story-page", { data: { story: story } }, {});
     assertContains('<link rel="amphtml" href="/amp/story/section%2Fslug"/>', string);
   });
 
+  it("does not append domain to amp visual stories if appendHostToAmpUrl not present", function () {
+    const story = { slug: "section/slug", "is-amp-supported": true };
+    const string = getSeoMetadata(seoConfig, config, "story-page", { data: { story: { ...story, "story-template": "visual-story" } } }, {});
+    assertContains('<link rel="amphtml" href="/section%2Fslug"/>', string);
+  });
+
```

Comment:
independent of story template, test is not needed

Reply:
removed